upup/pkg/fi/cloudup/terraform/target_hcl2.go

```diff
@@ -132,23 +133,34 @@ func (t *TerraformTarget) writeProviders(buf *bytes.Buffer) {
 		Write(buf, 0, fmt.Sprintf("provider %q", providerName))
 	buf.WriteString("\n")
 
-	// Add the second provider definition for managed files
-	if t.filesProvider != nil {
+	// Add any additional provider definition for managed files
+	keys := keysForMap(t.TerraformWriter.Providers)
+	sort.Strings(keys)
+	for _, key := range keys {
+		filesProvider := t.TerraformWriter.Providers[key]
```

Comment:
It's not necessarily the files provider at this point. Can this be named `provider`?

Reply:
Good call!